evap/staff/views.py

```diff
@@ -401,6 +401,20 @@ def semester_create(request):
     return render(request, "staff_semester_form.html", dict(form=form))
 
 
+@require_POST
+@manager_required
+def semester_make_active(request):
+    semester_id = request.POST.get("semester_id")
+    semester = get_object_or_404(Semester, id=semester_id)
+
+    Semester.objects.update(is_active=None)
+
+    semester.is_active = True
```

Comment:
I'd decorate the entire view with `@transaction.atomic` to eliminate race conditions (there's a brief period without an active semester)

Reply:
After a quick read of the documentation, I think we should just wrap the part after `get_object_or_404` into a `transaction.atomic` context, so that we don't need to get a lock for the database (or however `atomic` works here) if the request is invalid from the start.

What do you think?

Reply:
From the code perspective, the smaller context is a bit cleaner, so I'd go with that you can realize it with one additional line only (just as the decorator).

(I think it wouldn't really matter performance-wise, there's no real "locking" going on (as in: The whole db has to wait for this request). postgres is quite smart in being ACID while keeping good performance.)

@karyon good catch thinking about this, debugging might have been a nightmare here :D